go/vt/vtgate/vindexes/varchar_hash.go

```diff
@@ -0,0 +1,71 @@
+package vindexes
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+)
+
+// Varchar defines vindex that hashes an varchar to a KeyspaceId
+// by using bytes.toLower().
+type Varchar struct {
+	name string
+}
+
+// NewVarchar creates a new Varchar.
+func NewVarchar(name string, m map[string]interface{}) (Vindex, error) {
+	return &Varchar{name: name}, nil
+}
+
+// String returns the name of the vindex.
+func (vind *Varchar) String() string {
+	return vind.name
+}
+
+// Cost returns the cost of this index as 1.
+func (vind *Varchar) Cost() int {
+	return 1
+}
+
+// Verify returns true if id maps to ksid.
+func (vind *Varchar) Verify(_ VCursor, id interface{}, ksid []byte) (bool, error) {
+
+	data, err := getHash(id)
+	if err != nil {
+		return false, fmt.Errorf("Varchar_hash.Verify: %v", err)
+	}
+	return bytes.Compare(data, ksid) == 0, nil
+}
+
+func getHash(v interface{}) ([]byte, error) {
+	val, err := GetBytes(v)
+	return bytes.ToLower(val), err
+}
+
+// GetBytes returns bytes for a given value
+func GetBytes(key interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
```

Comment:
I don't think gob hashes the input. You can use this function instead: https://github.com/youtube/vitess/blob/master/go/vt/vtgate/vindexes/hash.go#L118. Should look something like this:

    func binHash(key interface{}) ([]byte, error) {
      source, ok := key.([]byte)
      if !ok {
        return nil, fmt.Errorf("unexpected data type for binHash: %T", key)
      }
      dest := make([]byte, len(source))
      block3DES.Encrypt(dest, source)
      return dest, nil
    }